src/engine/LoadingScreen.cs

```diff
@@ -104,6 +125,10 @@ public override void _Ready()
 
         spinner = GetNode<Control>("Spinner");
 
+        random = new Random();
+        randomizeTipTimer = new Timer { Interval = 5000, AutoReset = true };
+        randomizeTipTimer.Elapsed += (s, e) => RandomizeTip();
```

Comment:
You could test the timer, by making the in progress load just break out of the switch statement early so that it doesn't update the state. That way it should wait forever without blocking up the main thread.